pkgs/applications/editors/jetbrains/source/build.nix

```
@@ -190,33 +174,35 @@ let
   mkRepoEntry = entry: {
     name = ".m2/repository/" + entry.path;
     path = fetchurl {
-      urls = map (url: "${url}/${entry.url}") repositories;
+      urls = lib.flatten (
+        map (url: [
+          "https://cache-redirector.jetbrains.com/${url}/${entry.url}"
+          "https://${url}/${entry.url}"
+        ]) repositories
+      );
```

In this case, I think you can `concatMap` instead of `flatten (map)`.

In other cases, it may be better to reach for `concatLists` before resorting to `flatten`

```suggestion
      urls = lib.concatMap (url: [
        "https://cache-redirector.jetbrains.com/${url}/${entry.url}"
        "https://${url}/${entry.url}"
      ]) repositories;
```